pkg/agent/controller/bgp/controller.go

```diff
@@ -991,3 +983,94 @@ func (c *Controller) GetBGPPeerStatus(ctx context.Context) ([]bgp.PeerStatus, er
 	}
 	return peers, nil
 }
+
+// GetBGPRoutes returns the advertised/received routes to/from the given peerAddress.
+// If given peerAddress is empty, routes from all BGP Peers are returned.
+func (c *Controller) GetBGPRoutes(ctx context.Context, routeType bgp.RouteType, peerAddress string) (sets.Set[string], error) {
+	bgpPeerAddresses := c.getBGPPeerAddresses()
+	if bgpPeerAddresses == nil {
+		return nil, ErrBGPPolicyNotFound
+	}
+
+	// Validate peerAddress, consider empty peerAddress as valid.
+	if peerAddress != "" {
+		if !utilnet.IsIPv6String(peerAddress) && !utilnet.IsIPv4String(peerAddress) {
+			return nil, fmt.Errorf("invalid peer address: %s", peerAddress)
+		}
+		if !slices.Contains(bgpPeerAddresses, peerAddress) {
+			return nil, fmt.Errorf("bgp peer %s not found", peerAddress)
+		}
+		bgpPeerAddresses = []string{peerAddress}
+	}
+
+	bgpRoutes := make(sets.Set[string])
+	if routeType == bgp.RouteAdvertised {
+		// Advertised routes are same for all bgp peers.
+		bgpRoutesAdvertised := c.getBgpRoutesAdvertised()
```

Comment:
For an IPv4 peer, it will not "accept" advertised IPv6 routes; for an IPv6 peer, it will not "accept" advertised IPv4 routes either.  I think the cases are:

- `peerAddress` is not specified, return all routes.
- `peerAddress` is an IPv4 address, return IPv4 routes.
- `peerAddress` is an IPv6 address, return IPv6 routes.